promql/testdata/subquery.test

```diff
@@ -35,6 +38,10 @@ eval instant at 8000s rate(http_requests{group=~"pro.*"}[1m:10s])
   {job="api-server", instance="0", group="production"} 1
   {job="api-server", instance="1", group="production"} 2
 
+eval instant at 8000s rate(http_requests{group=~"pro.*"}[1m::9s])
```

Comment:
I would expect to see two tests which differ only in `:` vs `::`, to help nail down the expected differences.

Also probably a lot more tests in `subquery.test` and fewer in the `.go` files.